### PR TITLE
Inject Forge into Bukkit

### PR DESCRIPTION
**Pull request by m1lc0lm**

### CODE REVIEW

File: src/main/java/org/bukkit/Material.java [hunk 1]
```diff
@@ -3932,10 +3931,32 @@
...
+    // Magma/Mohist end
+
+    // Magma/Mohist start - constructor used to set if block is modded
+    private Material(final int id, boolean flag, String modName) {
+        this(id, 64);
+        this.isForgeBlock = flag;
+        this.modName = modName;
+        this.keyForge = new NamespacedKey(modName, this.name().toLowerCase(Locale.ROOT).substring(modName.length() + 1));
+    }
+    // Magma/Mohist end
```
Comment (Hexeption) [inline src/main/java/org/bukkit/Material.java:3958]: This is a no no

**Review by Hexeption — CHANGES_REQUESTED** [review]

File: src/main/java/org/bukkit/Material.java (hunk 1 above)
Comment (m1lc0lm) [reply src/main/java/org/bukkit/Material.java:3958]: I'll do an own solution

Comment (UeberallGebannt) [reply src/main/java/org/bukkit/Material.java:3958]: Might worth taking a look at Magma-1.16^^

Comment (m1lc0lm) [reply src/main/java/org/bukkit/Material.java:3958]: I already have ^^

**Pull request closed without merging**